ci/python-publish.yml

```diff
@@ -12,28 +12,42 @@ on:
   release:
     types: [published]
 
-permissions:
-  contents: read
-
 jobs:
-  deploy:
+  release-build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
 
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+
+      - name: build release distributions
+        run: |
+          # NOTE: put your own distribution build steps here.
+          python -m pip install build
+          python -m build
+
+      - name: upload distributions
+        uses: actions/upload-artifact@v4
+        with:
+          name: release-dists
+          path: dist/
+
+  pypi-publish:
     runs-on: ubuntu-latest
+    needs:
+      - release-build
+    permissions:
+      id-token: write
 
     steps:
-    - uses: actions/checkout@v3
-    - name: Set up Python
-      uses: actions/setup-python@v3
-      with:
-        python-version: '3.x'
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install build
-    - name: Build package
-      run: python -m build
-    - name: Publish package
-      uses: pypa/gh-action-pypi-publish@27b31702a0e7fc50959f5ad993c78deac1bdfc29
-      with:
-        user: __token__
-        password: ${{ secrets.PYPI_API_TOKEN }}
+      - name: Retrieve release distributions
+        uses: actions/download-artifact@v4
+        with:
+          name: release-dists
+          path: dist/
+
+      - name: Publish release distributions to PyPI
```

Comment:
Master script??

Reply:
52 yes

Reply:
This might be a dialect thing: I've only ever heard "PyPI" without the *the* definite article, whereas expanded it would be "*the* Python Package Index" 🙂

Reply:
It's a grammar thing (think: *the* US). Though I indeed saw various usage patterns as well. OTOH PyPI is unique enough not to need an article (people don't tend to confuse it with Test PyPI).

I'm likely confusing it with the preferred usage of "the PyPA". I did some googling and only found it the long form: https://packaging.python.org/en/latest/glossary/#term-pypi.org. The FAQ on pypi.org doesn't use the article in front of the abbreviation.

https://english.stackexchange.com/a/82827/68752 suggests this depends on how one would pronounce an abbreviation.

Anyway, I'm going to trust your instinct. Please, disregard this suggestion.